mobile/library/common/internal_engine.cc

```diff
@@ -285,7 +285,7 @@ void InternalEngine::onDefaultNetworkAvailable() {
   ENVOY_LOG_MISC(trace, "Calling the default network available callback");
 }
 
-void InternalEngine::onDefaultNetworkChanged(NetworkType network) {
+void InternalEngine::onDefaultNetworkTypeChanged(NetworkType network) {
```

Comment:
IMO, no need to rename it. We eventually want this callback to be triggered not just by network type, but also the network itself.

Reply:
I think it's probably better to have a different callback for that. This makes it similar to how it's implemented in Chromium: https://source.chromium.org/chromium/chromium/src/+/main:net/android/java/src/org/chromium/net/NetworkChangeNotifierAutoDetect.java;l=974

Reply:
The caller of the chromium interface also calls it when there is non-type change though: https://source.chromium.org/chromium/chromium/src/+/main:net/android/java/src/org/chromium/net/NetworkChangeNotifierAutoDetect.java;l=1390-1396;drc=8e948282d37c0e119e3102236878d6f4d5052c16;bpv=1;bpt=1

Reply:
Right, but the callback name is `onConnectionTypChanged` and not `onConnectionChanged`: https://source.chromium.org/chromium/chromium/src/+/main:net/android/java/src/org/chromium/net/NetworkChangeNotifierAutoDetect.java;l=974;drc=8e948282d37c0e119e3102236878d6f4d5052c16

Especially for now we are only interested in the changes in the network type and not on all changes in the network capabilities, hence the rename. But if you feel strongly about it, I can change it back. Let me know.

Reply:
We don't need to exactly follow the naming of Chromium code, especially when Chromium naming is confusing. If you don't feel like to change it back, I can do it in a follow up PR.

Reply:
Done. I reverted the rename.